chore: migration guide for v10 (#1419)
AmadeusITGroup/otter: migrations/10.0.md

# Major 10.0 Migration guide
> [!IMPORTANT]
> Please make sure to first follow the ["Upgrade new version guide"](https://github.com/AmadeusITGroup/otter/blob/release/main/docs/core/UPGRADE_NEW_VERSION.md) before going through these steps.

> ## Customization

Reviewer: ```suggestion
## Personalization
```

> Customization now only relies on the ```@O3rComponent``` decorator.<br>

Reviewer: ```suggestion
Now, component type must be set through ```@O3rComponent``` decorator.
This decorator is also used to identify components in our [Chrome extension](link/to/the/doc).<br>
```

> The following interfaces and decorator (already deprecated) have now been fully removed.

Reviewer 2: ```suggestion
The following interfaces and decorator (already deprecated) have been fully removed.
```

> * ``@InputMerge``

Reviewer 2: ```suggestion
* ``@InputMerge`` decorator
```

> * ``@InputMerge`` decorator

Reviewer: Maybe we could have a paragraph that referenced all removed methods, interfaces, decorators, ...
Because I'm not sur `@InputMerge` is linked to personalization
(and we can remove it from authorized decorator in the eslint config `packages/@o3r/eslint-config-otter/rules/typescript/eslint.cjs L85`

* ``Blocks``, ``Page``, ``ExposedComponent`` interfaces

> These change have already been automated in the v9 ng updates.

Reviewer: ```suggestion
These changes have already been automated in the v9 ng updates.
```

## utils.Date
* ``dateInstance`` has been removed as it was only returning the instance of the object
* ``getUtilsDate`` used to create a ``utils.Date`` instance from a ``utils.DateTime`` instance has been removed.<br>
Please use instead:
```typescript
const myUtilsDateTimeInstance = new utils.DateTime();
const myDate = new utils.Date(myUtilsDateTimeInstance)
```

## Element fixtures

> ``throwOnUndefined``'s signature has been updated and now returns a Promise.

Reviewer 2: ```suggestion
The signature of ``throwOnUndefined`` has been updated and now returns a Promise.
```

## End of @ama-sdk/generator-sdk

> ``@ama-sdk/generator-sdk``, based on yeoman and the Swagger CodeGen and deprecated since v8, is now completely gone and replaced by ``@ama-sdk/schematics`` running with angular schematics and the OpenApiTools generator.

Reviewer 2: ```suggestion
``@ama-sdk/generator-sdk``, based on yeoman and the Swagger CodeGen and deprecated since Otter v8, is now completely gone and replaced by ``@ama-sdk/schematics`` running with Angular schematics and the OpenApiTools generator.
```

> ``@ama-sdk/generator-sdk``, based on yeoman and the Swagger CodeGen and deprecated since Otter v8, is now completely gone and replaced by ``@ama-sdk/schematics`` running with Angular schematics and the OpenApiTools generator.

Reviewer: ```suggestion
``@ama-sdk/generator-sdk``, based on yeoman and the Swagger CodeGen is deprecated since Otter v8, is now completely gone and replaced by ``@ama-sdk/schematics`` running with Angular schematics and the OpenApiTools generator.
```

> If you have not yet migrated, the migration to ``@ama-sdk/schematics`` is automated since v9:

Reviewer 2: ```suggestion
If you have not yet done so, the migration to ``@ama-sdk/schematics`` is automated since v9:
```

```shell
# In an angular context
ng add @ama-sdk/schematics

# yarn usage
yarn add -D @ama-sdk/schematics

> yarn schematics @ama-sdk/schematics:ng-add

Reviewer 3: regarding the `npx` example below, it should not be
```suggestion
yarn dlx -p @angular-devkit/schematics-cli schematics @ama-sdk/schematics:ng-add
```

or, if you want to support yarn1:
```suggestion
yarn add -D @angular-devkit/schematics-cli
yarn schematics @ama-sdk/schematics:ng-add
```
?

# npm usage

> npm install --save-dev @ama-sdk/schematics

Reviewer: ```suggestion
npm install --save-dev @ama-sdk/schematics @angular-devkit/schematics-cli
```
for the same reason as above

> npx schematics @ama-sdk/schematics:ng-add

Reviewer 4: ```suggestion
npx -p @angular-devkit/schematics-cli schematics @ama-sdk/schematics:ng-add
```
